File: python/easy/581.py
class Solution:
    def findUnsortedSubarray(self, nums: List[int]) -> int:
        n = len(nums)
        l,r = 0,n-1
        big,small = nums[0],nums[-1]
        for i in range(1,n):
            if nums[i] < big:
                l = max(l,i)
            else:
                big = nums[i]
        for i in range(n-1,-1,-1):
            if nums[i] > small:
                r = min(r,i)
            else:
                small = nums[i]
        if r-l+1 == n: return 0
        return l-r+1

File: go/581.go
package leetcode

func findUnsortedSubarray(nums []int) int {
    n := len(nums)
    l,r := 0,n-1
    max,min := nums[0],nums[n-1]
    for i :=1;i<n;i++ {
        if nums[i] >= max {
            max = nums[i]
        } else {
            l = i
        }
    }
    for i:=n-1;i>=0;i-- {
        if nums[i] > min {
            r = i
        } else {
            min = nums[i]
        }
    }
    if r - l + 1 == n {
        return 0
    } else {
        return l - r + 1
    }
}